here-naksha-lib-base/src/commonMain/kotlin/Flags.kt

```diff
@@ -0,0 +1,70 @@
+@file:OptIn(ExperimentalJsExport::class)
+
+package com.here.naksha.lib.nak
+
+import kotlin.js.ExperimentalJsExport
+import kotlin.js.JsExport
+
+/**
+ * Keeps flags on different bits of integer
+ * GE - geometry encoding - bits: 0-5
+ * FE - feature encoding - bits: 6-11
+ * RE - reserved for future - bits: 12-31
+ *
+ *       12-31 RE        6-11 FE  0-5 GE
+ * [00000000000000000000][000000][00000]
+ */
+@JsExport
+class Flags(initialFlags: Int? = null) {
+    private var featureEncoding = initialFlags?.let { (it and 0b0000000000000000000_111111_000000).shr(6) } ?: DEFAULT_FEATURE_ENCODING
+    private var geometryEncoding = initialFlags?.let { it and 0b0000000000000000000_000000_111111 } ?: DEFAULT_GEOMETRY_ENCODING
+    private var reserved = initialFlags?.ushr(12) ?: 0
+
+    fun getReserved() = reserved
+
+    fun getGeometryEncoding() = geometryEncoding
+
+    fun getFeatureEncoding() = featureEncoding
+
+    fun setGeometryEncoding(ge: Int) {
+        check(ge < 64) // 2^6-1
```

Comment:
You should test against negative values too, so `require(ge in 0..63)` (in this case I would argue require is better, so better IllegalArgumentException than IllegalStateException).